sys/kern/src/syscalls.rs

```diff
@@ -59,7 +59,7 @@ use crate::umem::{safe_copy, USlice};
 /// the way in and restored on the way out, (2) call this from the syscall
 /// interrupt handler, only, and (3) not call it reentrantly.
 #[no_mangle]
-pub unsafe extern "C" fn syscall_entry(nr: u32, task: *mut Task) {
+pub unsafe extern "C" fn syscall_entry(nr: u32, task: *const Task) {
```

Comment:
When did the requirement for this `mut` change?

Reply:
This is purely based on reading of the code. I am not sure why this was `mut` originally. Need to follow up on the history to check this.

Reply:
Going back to the place where this was originally added in 31273f783185c115ba8d52520cd4d0a1901e051c, I don't see a reason why this was marked as mut. It is used only for calculating the index of the current task in the task table. So, I am guessing it should be safe to drop the mut. Also, the compiler didn't complain for any of the architectures, so I am assuming it should be fine?

Reply:
There is this comment which I am not completely sure about: https://github.com/rivosinc/hubris/commit/31273f783185c115ba8d52520cd4d0a1901e051c#diff-e7c79d8a70702b5e115b3efa9ef914c7c7af40d78ff0e99c3bf728d32a8d440dR329

```
// Use the task pointer, which now aliases a `&mut` slice and shall not be
    // dereferenced, into a task index. Yeah, we could store the task index
    // alongside the pointer. Maybe later.
```
Was it marked as mut because the tasks table itself is mut and task aliases a mut slice? Do we lose anything by making it const? @zachreizner to see if he can provide some inputs.

Reply:
Thanks @zachreizner! Going ahead and marking this as resolved.